test/forms/buttons/button.test.js

```diff
@@ -1,136 +1,167 @@
     </Button>
   )
-  wrapper.find('button').simulate('click')
+
+  const user = userEvent.setup()
+  const button = screen.getByRole('button')
+  await user.click(button)
 
   expect(onClick).not.toHaveBeenCalled()
 })
 
-test('Button onClick is not run when form is pristine', () => {
+test('Button onClick is not run when form is pristine', async () => {
   const onClick = jest.fn()
-  const wrapper = mount(
+  render(
     <Button onClick={onClick} pristine={true}>
-      {' '}
       Hi
     </Button>
   )
-  wrapper.find('button').simulate('click')
+
+  const user = userEvent.setup()
+  const button = screen.getByRole('button')
+  await user.click(button)
 
   expect(onClick).not.toHaveBeenCalled()
 })
 
-test('Button onClick is not run when form is submitting', () => {
+test('Button onClick is not run when form is submitting', async () => {
   const onClick = jest.fn()
-  const wrapper = mount(
+  render(
     <Button onClick={onClick} submitting={true}>
-      {' '}
       Hi
     </Button>
   )
-  wrapper.find('button').simulate('click')
+
+  const user = userEvent.setup()
+  const button = screen.getByRole('button')
+  await user.click(button)
 
   expect(onClick).not.toHaveBeenCalled()
 })
 
 test('Button adds variant string to class', () => {
-  const wrapper = shallow(<Button variant="custom"> Hi</Button>)
-  expect(wrapper.hasClass('button-custom')).toBe(true)
+  render(<Button variant="custom">Hi</Button>)
+  const button = screen.getByRole('button')
+
+  expect(button).toHaveClass('button-custom')
 })
 
 test('Button adds type to button', () => {
-  const wrapper = shallow(<Button type="reset"> Hi</Button>)
-  expect(wrapper.props().type).toBe('reset')
+  render(<Button type="reset">Hi</Button>)
+  const button = screen.getByRole('button')
+
+  expect(button).toHaveAttribute('type', 'reset')
 })
 
 test('Button has class "in-progress" when form is submitting', () => {
-  const wrapper = shallow(<Button submitting={true}> Hi</Button>)
-  expect(wrapper.hasClass('in-progress')).toBe(true)
+  render(<Button submitting={true}>Hi</Button>)
+  const button = screen.getByRole('button')
+
+  expect(button).toHaveClass('in-progress')
 })
 
 test('Button passes extra props to button element', () => {
-  const onClick = () => 'yo'
-  const wrapper = shallow(<Button onClick={onClick}> Hi</Button>)
-  expect(wrapper.props().onClick).toBe(onClick)
+  render(<Button test="test">Hi</Button>)
+  const button = screen.getByRole('button')
+
+  expect(button).toHaveAttribute('test')
 })
 
 test('Specifying a class name prop does not override variant class', () => {
-  const wrapper = shallow(
+  render(
     <Button variant="primary" className="button-large">
       Click Me
     </Button>
   )
-  expect(wrapper.hasClass('button-primary')).toBe(true)
-  expect(wrapper.hasClass('button-large')).toBe(true)
+  const button = screen.getByRole('button')
+
+  expect(button).toHaveClass('button-primary')
+  expect(button).toHaveClass('button-large')
```

Comment:
`toHaveClass` accepts a list as well

https://github.com/testing-library/jest-dom#tohaveclass
```suggestion
  expect(button).toHaveClass('button-primary', 'button-large')
```